crates/node/src/cli/l1.rs

```diff
@@ -0,0 +1,20 @@
+use url::Url;
+
+fn parse_url(s: &str) -> Result<Url, url::ParseError> {
+    s.parse()
+}
+
+#[derive(Clone, Debug, clap::Args)]
+pub struct L1SyncParams {
+    /// Disable L1 sync.
+    #[clap(long, alias = "no-l1-sync")]
+    pub sync_l1_disabled: bool,
+
+    /// The L1 rpc endpoint url for state verification.
+    #[clap(long, value_parser = parse_url, value_name = "ETHEREUM RPC URL")]
+    pub l1_endpoint: Option<Url>,
```

Comment:
you can enforce in clippy that `sync_l1_disabled` false requires `l1_endpoint` afaik

Reply:
resolved